dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -62,14 +62,53 @@ static const columninfo RecycleBinColumns[] =
  * Recycle Bin folder
  */
 
-HRESULT CRecyclerExtractIcon_CreateInstance(LPCITEMIDLIST pidl, REFIID riid, LPVOID * ppvOut)
+BOOL WINAPI CBSearchRecycleBin(IN PVOID Context, IN HANDLE hDeletedFile);
+
+static PIDLRecycleStruct * _ILGetRecycleStruct(LPCITEMIDLIST pidl);
+
+typedef struct _SEARCH_CONTEXT
 {
+    PIDLRecycleStruct *pFileDetails;
+    HANDLE hDeletedFile;
+    BOOL bFound;
+} SEARCH_CONTEXT, *PSEARCH_CONTEXT;
+
+HRESULT CRecyclerExtractIcon_CreateInstance(
+    LPCITEMIDLIST pidl, REFIID riid, LPVOID * ppvOut)
+{
+    PIDLRecycleStruct *pFileDetails = _ILGetRecycleStruct(pidl);
+    if (pFileDetails == NULL)
+        goto error;
+
+    // Try to obtain the file
+    SEARCH_CONTEXT Context;
+    Context.pFileDetails = pFileDetails;
+    Context.bFound = FALSE;
+
+    EnumerateRecycleBinW(NULL, CBSearchRecycleBin, (PVOID)&Context);
+    if (Context.bFound)
+    {
+        // This should be executed any time, if not, there are some error in the implementation
```

Comment:
Nitpick: Either there _is_, or _errors_.

Reply:
erros, i missed the s

Reply:
Done :D